traffic_portal/app/src/modules/public/login/LoginController.js

```diff
@@ -48,9 +53,28 @@ var LoginController = function($scope, $log, $uibModal, authService, userService
         });
     };
 
+    $scope.loginOauth = function() {
+        const gotoUrl = new URL(propertiesModel.properties.oAuth.oAuthUrl);
+        const currentUrl = new URL(window.location.href);
+        const afterHash = currentUrl.hash.substring(2);
+        const urlNoHash = new URL(currentUrl.protocol + '//' + currentUrl.hostname + currentUrl.pathname + afterHash);
+
+        const redirectParam = urlNoHash.searchParams.get('redirect') !== null ? urlNoHash.searchParams.get('redirect') : '';
+        var continueParam = new URL(window.location.href);
+        continueParam.hash = '#!/sso?redirect=' + redirectParam;
+
+        var continueURL = new URL(gotoUrl.protocol + '//' + gotoUrl.hostname + gotoUrl.pathname);
```

Comment:
this is all pretty confusing. can you add some comments to explain the format of the url you are building to send them to?

Reply:
done